src/Action.hs

```diff
@@ -69,13 +77,35 @@ import Test.QuickCheck (arbitrary, generate)
 
 -- | User representation during an action
 -- FIXME: Figure out which information is needed here.
-data UserState
-    = UserLoggedOut
-    | UserLoggedIn { _username :: UserLogin, _sessionCookie :: ST }
+data UserState = UserState
+    { _usSessionToken :: Maybe ThentosSessionToken
```

Comment:
What is this used for?  Since user id is stored here, too, I think we don't need the session token for authentication or authorization.  Does thentos require it for the `GetThentosSessionToken` instance?